packages/dds/tree/docs/wip/schema-evolution.md

```diff
@@ -0,0 +1,583 @@
+Before they know it, a significant part of their schema has been recreated as the new data format "bubbles up" the tree.
+
+> In this way, if a schema node changes, then all schema nodes that transitively reference/contain that node will also have to change.
+  Therefore a typical application schema will observe a "spine" of its schema tree change when _anything_ in the schema changes, climbing up to the application's root node.
+  Since the root node always changes, even a very small change to a leaf schema node results in a "replacement" of the entire schema.
+
+Migrating the application from the old schema to the new schema can be difficult.
+The next section will detail the worst case scenario, but subsequent sections will explore how the migration might be [made easier for some of the types of changes listed above](#schema-change-shortcuts).
+
+### Migration
+
+Many developers who change their schema will need to think carefully about how they migrate their application from the old schema to the new schema.
+It requires thoughtful planning in a production scenario.
+
+An application has its schema encoded in multiple places.
+
+* The **stored schema** is the schema that the tree in the **document** - the persisted state and source of truth for the application's data, shared by all clients - conforms to.
+  It is persisted as part of the document.
+  All clients reading the document know that the tree stored in that document matches this schema.
+* The **view schema** is the schema that a client expects when reading and writing tree.
+  It is part of the application source code loaded by a client.
+
+When an application is first written, these two schema are equivalent.
+In some environments, the developer can ensure that they remain equivalent as new versions are released.
+For example, consider an application where all the documents are stored on a central server.
+To release a new version, a developer takes the following steps to upgrade from schema A to schema B:
+
+1. At first, the documents are using stored schema A and the application is written against stored schema A.
+2. The developer temporarily takes the service offline, suspending application availability, and then upgrades all documents on the server to use stored schema B before re-enabling the service.
+   Thereafter, any clients attempting to connect to the service will be prompted to download the new version of the application client first, which is written against view schema B.
+3. The migration is complete.
+
+That makes a clean and relatively simple transition.
+The diagram below visualizes the clean break between the old and new schema that happens at step 2:
+
+```mermaid
+gantt
+dateFormat D
+axisFormat Step %e
+tickInterval 1day
+Maintain view schema A in application source code : 1, 1d
+Documents contain stored schema A                 : 1, 1d
+Maintain view schema B in application source code : 2, 1d
+Documents contain stored schema B                 : 2, 1d
+```
```

Comment:
Your diagram missing the step in between that should read "application is not available, please come back in 48 hours" :)